### DISCUSSION TITLE
How to print the evolution of mipgap through iterations?

### DISCUSSION THREAD

**Discussion by boiro9** [coin-or/Cbc#505] [Q&A] [answered]
Is there any option to print the evolution of mipgap through the iterations in the output while CBC is running?
I have tried the following options:
{'ratio': 1,  'slog': 3}
and it prints now many information but the gap is not included throught the iterarions, I can only see the final gap of the solution.

**Comment by jjhforrest** [comment, marked as answer]
Why do you want this?


You get a feel for how the gap is changing just from the normal output e.g.

Cbc0010I After 1000 nodes, 9 on tree, 7350 best solution, best possible 
6848.5558 (3.88 seconds)

and a gap of approximately 7%

If you want to stop when the gap drops beow 7% then you can set 
-ratiogap 0.07

Otherwise choose an example e.g. interrupt.cpp and modify the event 
handler to something like

    } else if (whichEvent == treeStatus) {
       if (model_->bestSolution()) {
     double bestPossible = model_->getBestPossibleObjValue();
     double bestObjective = model_->getObjValue();
     printf("Gap %g\n",(bestObjective-bestPossible)/fabs(bestPossible));
       } else {
     printf("No solution found yet\n");
       }
       return noAction; // carry on
